062-support-tiered-storage.md

```diff
@@ -0,0 +1,131 @@
+    rlmm.config.remote.log.metadata.common.client.security.protocol: SSL
+    rlmm.config.remote.log.metadata.topic.replication.factor: 1
+```
+
+The configuration above will get translated to the below Kafka broker config:
+```
+# RLMM configuration generated by Strimzi
+remote.log.storage.system.enable: true
+remote.log.metadata.manager.impl.prefix: rlmm.config.
+remote.log.metadata.manager.class.name: org.apache.kafka.server.log.remote.metadata.storage.TopicBasedRemoteLogMetadataManager
+#RSM configs set by the operator and by the user
+remote.log.storage.manager.class.name: com.example.kafka.tiered.storage.s3.S3RemoteStorageManager
+remote.log.storage.manager.class.path: /opt/kafka/plugins/tiered-storage-s3/*
+remote.log.storage.manager.impl.prefix: rsm.config.
+rsm.config.storage.bucket.name=my-bucket
+# # Custom RLMM configs
+rlmm.config.remote.log.metadata.common.client.bootstrap.servers: broker-bootstrap.com:9091
+rlmm.config.remote.log.metadata.common.client.security.protocol: SSL
+rlmm.config.remote.log.metadata.topic.replication.factor: 1
+```
+
+### Validation
+
+There are certain [limitations](https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Tiered+Storage+Early+Access+Release+Notes) for a tiered storage enabled cluster.
+
+We choose not to validate the error conditions at the Strimzi operator layer because:
+
+* At least some of them are expected to be removed in the future
+* Validating them might be quite complex (for example, IIRC, the compacted topics can be set in the topic itself as well as in the broker defaults)
+* Validating this in UTO would capture only a part of the topics and not prevent the users from misconfiguration. So they will still need to be aware of these rules and learn to respect them.
+
+### Docker image
+
+This proposal doesn't include built in plugin support in any Strimzi built Kafka image. Users need to specify the path via the config, and customize the docker image to include the library.
+
+An example to build such custom image is via a custom Dockerfile:
+
+```
+FROM quay.io/strimzi/kafka:0.40.0-kafka-3.6.0
+ARG REMOTE_PLUGIN_PATH
+COPY ${REMOTE_PLUGIN_PATH} /opt/kafka/plugins/tiered-storage-s3/
+```
+
+An image with plugin library can be built with the following command:
+
+```
+docker build -t docker.example.com/tierstorage/kafka:0.40.0-kafka-3.6.0-plugin-1 . -build-arg REMOTE_PLUGIN_PATH=./s3/build/install/s3/*
+```
+## Testing strategy
+
+Unit tests will be added to validate that all the RLM, RSM, and RLMM changes are propagated through.
+Because the RSM and RLMM are typically provided by user and there is no built in support for any third party cloud provider, there won’t be any integration tests covering the cloud provider interaction. It’s the user’s responsibility to ensure the RSM and RLMM works with the intended cloud provider.
+
+For testing purposes, an integration test with `LocalTieredStorage` implementation can be added to ensure end-to-end functionality.
```

Comment:
I guess it would be a system test in a Strimzi terminology.

```suggestion
For testing purposes, an system test with `LocalTieredStorage` implementation can be added to ensure end-to-end functionality.
```